### ISSUE TITLE
[vault-configurer] nil point at `getExistingGroups`

### ISSUE THREAD

**Issue by Gentoli** [banzaicloud/bank-vaults#1546]
**Describe the bug**:
`vault-configurer` panics with `invalid memory address or nil pointer dereference`

**Expected behaviour**:
Nothing crashes

**Steps to reproduce the bug**:
Not very sure.. there is a stacktrace for the panic.
Removing `purgeUnmanagedConfig` stopped the crashloop
```
    purgeUnmanagedConfig:
      enabled: true
      exclude:
        secrets: true
```

**Additional context**:

**Environment details**:
- Kubernetes version (e.g. v1.10.2):
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:38:50Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4-gke.1501", GitCommit:"efcab47cafa8f8cb6850be0a379247020eb3c33b", GitTreeState:"clean", BuildDate:"2022-01-04T20:28:48Z", GoVersion:"go1.16.10b7", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): `GKE`
- bank-vaults version (e.g. 0.4.17): `1.15.0`
- Install method (e.g. helm or static manifests): `operator (chart 1.15.2, deployed by argocd)`
- Logs from the misbehaving component (and any other relevant logs):
<details>
  <summary>Click to expand!</summary>
  
  ```logs
{"level":"info","msg":"vault metrics exporter enabled: :9091/metrics","time":"2022-02-22T03:45:58Z"}
{"level":"info","msg":"applying config file : /config/vault-configurer/vault-config.yml","time":"2022-02-22T03:45:58Z"}
{"level":"info","msg":"checking if vault is sealed...","time":"2022-02-22T03:45:58Z"}
{"level":"info","msg":"watching directory for changes: /config/vault-configurer/","time":"2022-02-22T03:45:58Z"}
{"level":"info","msg":"vault is unsealed, configuring...","time":"2022-02-22T03:45:58Z"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x14fb439]

goroutine 1 [running]:
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).getExistingGroups(0xc00001d890)
	/build/internal/vault/identity_groups.go:151 +0x139
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).removeUnmanagedGroups(0xc00001d890, {0x0, 0x0, 0x0})
	/build/internal/vault/identity_groups.go:211 +0x70
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).configureIdentityGroups(0xc00001d890)
	/build/internal/vault/identity_groups.go:342 +0xae
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).Configure(0xc00001d890, 0x4)
	/build/internal/vault/operator_client.go:571 +0x9de
main.glob..func2.2({0x25aa440, 0xc00001d890}, 0xc00063fcc8, 0xc000a0fc40, 0x0, 0xc00009ade0, 0xc0000555f0)
	/build/cmd/bank-vaults/configure.go:127 +0x23b
main.glob..func2(0x3722560, {0x2111d09, 0xf, 0xf})
	/build/cmd/bank-vaults/configure.go:147 +0x54c
github.com/spf13/cobra.(*Command).execute(0x3722560, {0xc0000d00f0, 0xf, 0xf})
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x37222e0)
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
main.execute()
	/build/cmd/bank-vaults/main.go:138 +0xde
main.main()
	/build/cmd/bank-vaults/main.go:278 +0x5c
  ```
</details>

- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:

/kind bug

**Comment by aabouzaid** [comment]
Thanks @Gentoli for reporting this issue.
I believe that this issue has been fixed by #1538
Please test it again with bank-vaults `v1.15.1`.

**Comment by Gentoli** [comment]
Yes I don't see this error anymore in `v1.15.1`. Thanks!
